crates/biome_configuration/src/analyzer/assists/rules.rs

```diff
@@ -0,0 +1,152 @@
+    #[deserializable(rename = "source")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub source: Option<Source>,
+}
+impl Rules {
+    #[doc = r" Checks if the code coming from [biome_diagnostics::Diagnostic] corresponds to a rule."]
+    #[doc = r" Usually the code is built like {group}/{rule_name}"]
+    pub fn has_rule(group: RuleGroup, rule_name: &str) -> Option<&'static str> {
+        match group {
+            RuleGroup::Source => Source::has_rule(rule_name),
+        }
+    }
+    #[doc = r" Given a category coming from [Diagnostic](biome_diagnostics::Diagnostic), this function returns"]
+    #[doc = r" the [Severity](biome_diagnostics::Severity) associated to the rule, if the configuration changed it."]
+    #[doc = r" If the severity is off or not set, then the function returns the default severity of the rule:"]
+    #[doc = r" [Severity::Error] for recommended rules and [Severity::Warning] for other rules."]
+    #[doc = r""]
+    #[doc = r" If not, the function returns [None]."]
+    pub fn get_severity_from_code(&self, category: &Category) -> Option<Severity> {
+        let mut split_code = category.name().split('/');
+        let _lint = split_code.next();
+        debug_assert_eq!(_lint, Some("assists"));
+        let group = <RuleGroup as std::str::FromStr>::from_str(split_code.next()?).ok()?;
+        let rule_name = split_code.next()?;
+        let rule_name = Self::has_rule(group, rule_name)?;
+        match group {
+            RuleGroup::Source => self
+                .source
+                .as_ref()
+                .and_then(|group| group.get_rule_configuration(rule_name))
+                .filter(|conf| !matches!(conf, RuleAssistConfiguration::Off))
+                .map(|conf| conf.into()),
+        }
+    }
+    #[doc = r" It returns the enabled rules by default."]
+    #[doc = r""]
+    #[doc = r" The enabled rules are calculated from the difference with the disabled rules."]
+    pub fn as_enabled_rules(&self) -> FxHashSet<RuleFilter<'static>> {
+        let mut enabled_rules = FxHashSet::default();
+        let mut disabled_rules = FxHashSet::default();
+        if let Some(group) = self.source.as_ref() {
+            enabled_rules.extend(&group.get_enabled_rules());
+            disabled_rules.extend(&group.get_disabled_rules());
+        }
+        enabled_rules.difference(&disabled_rules).copied().collect()
+    }
+}
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, Merge, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(JsonSchema))]
+#[serde(rename_all = "camelCase", default, deny_unknown_fields)]
+#[doc = r" A list of rules that belong to this group"]
+pub struct Source {
+    #[doc = r" It enables the recommended rules for this group"]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub recommended: Option<bool>,
+    #[doc = r" It enables ALL rules for this group."]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub all: Option<bool>,
```

Comment:
Are you finally keeping the `recommended`/`all` options?

Reply:
Oh, that's the codegen, thank you for the double check. It wasn't intended